GoogleForCreators/web-stories-wp
# Improve AMP sanitization for title tags and meta descriptions

## Context



We had issues with missing title tags / meta descriptions or duplicate ones, depending on the theme and SEO plugin being used.

I also contributed a [PR to Yoast SEO](https://github.com/Yoast/wordpress-seo/pull/19214) to improve their logic, as it turns out they did some odd changes there recently.

## Summary



Prevent missing title tags / meta descriptions or duplicate ones at the AMP sanitization level.

Also added another fix to ensure links always have a proper protocol.

## Relevant Technical Choices



## To-do



## User-facing changes



No visual changes

## Testing Instructions






This PR can be tested by following these steps:

1. View a published story on the frontend
2. Verify that there's a `<title>` tag and a meta description present

Test with:

* Twenty Twelve theme
* Twenty Twenty-Three theme
* OceanWP theme
* Yoast SEO enabled/disabled

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #12655
Fixes #12709
